drivers/hmis/app/graphql/types/hmis_schema/has_enrollments.rb

```diff
@@ -12,12 +12,14 @@ module HasEnrollments
       extend ActiveSupport::Concern
 
       class_methods do
-        def enrollments_field(name = :enrollments, description = nil, type: Types::HmisSchema::Enrollment.page_type, **override_options, &block)
+        def enrollments_field(name = :enrollments, description = nil, type: Types::HmisSchema::Enrollment.page_type, without_args: [], **override_options, &block)
           default_field_options = { type: type, null: false, description: description }
           field_options = default_field_options.merge(override_options)
           field(name, **field_options) do
             argument :sort_order, HmisSchema::EnrollmentSortOption, required: false
             argument :include_in_progress, GraphQL::Types::Boolean, required: false
+            argument :project_types, [Types::HmisSchema::Enums::ProjectType], required: false  unless without_args.include? :project_types
```

Comment:
I'd love to se "without_args" flipped to a positive, it's often easier to write the negative case to exclude something that otherwise is there, but it's really hard to follow `unless without include`.  Could it be something like:
```
def enrollments_field(name = :enrollments, description = nil, type: Types::HmisSchema::Enrollment.page_type, extra_args: [:project_types], **override_options, &block)
....
argument :project_types, [Types::HmisSchema::Enums::ProjectType], required: false if extra_args.include? :project_types
```

Reply:
or maybe better if it's just 
```
def enrollments_field(name = :enrollments, description = nil, type: Types::HmisSchema::Enrollment.page_type, included_args: [:sort_order, :include_in_progress, :project_types, :client_search_term], **override_options, &block)
```
and then all of the args have a check.
I'm not sold on either of those either.

Reply:
Good point, but I'm not sure yet if it's worth the change. I think the way it works often _is_ exclusionary in nature... like you add a new search term (say `client_search_term` for Enrollment) and there's ONE place it really doesn't make sense to use (resolving within the context of one client like `Client.enrollments(...)`) but everywhere else (`Project.enrollments(...)`, `Organization.enrollments(...)`, `enrollments(...)`) it DOES make sense. If that's a common pattern, I'd rather just exclude one  place than have to add it in everywhere, I think. Let's wait and see how it plays out to see whether it should be flipped from a negative to positive.

Reply:
Sounds good.